src/main/java/uk/gov/ons/ssdc/caseprocessor/cache/UacQidCache.java

```diff
@@ -72,25 +66,21 @@ public void afterCompletion(int status) {
     }
   }
 
-  private void topUpQueue(int questionnaireType) {
-    synchronized (isToppingUpQueue) {
-      if (!isToppingUpQueue.contains(questionnaireType)
-          && uacQidLinkQueueMap.get(questionnaireType).size() < cacheMin) {
-        isToppingUpQueue.add(questionnaireType);
+  private void topUpCache() {
+    synchronized (lock) {
```

Comment:
Could this be synchronised on `isToppingUpQueue` rather than creating a stand in empty object?

Reply:
So that's what I had originally but it's considered bad practice to use synchronised on a boolean. Here's a Baeldung article on it https://www.baeldung.com/java-synchronization-bad-practices#booleanLiteral . Happy to discuss it though around another way of doing it

Reply:
Good to know, this way of doing it makes sense then 👍